test/e2e/gameserver_test.go

```diff
@@ -736,3 +736,161 @@ func TestGameServerResourceValidation(t *testing.T) {
 	assert.Equal(t, metav1.CauseTypeFieldValueInvalid, statusErr.Status().Details.Causes[0].Type)
 	assert.Equal(t, "container", statusErr.Status().Details.Causes[0].Field)
 }
+
+func TestGameServerSetPlayerCapacity(t *testing.T) {
+	if !runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	t.Run("no initial capacity set", func(t *testing.T) {
+		gs := framework.DefaultGameServer(defaultNs)
+		gs, err := framework.CreateGameServerAndWaitUntilReady(defaultNs, gs)
+		if err != nil {
+			t.Fatalf("Could not get a GameServer ready: %v", err)
+		}
+		assert.Equal(t, gs.Status.State, agonesv1.GameServerStateReady)
+		assert.Equal(t, int64(0), gs.Status.Players.Capacity)
+
+		reply, err := e2eframework.SendGameServerUDP(gs, "PLAYER_CAPACITY")
+		assert.NoError(t, err)
+		assert.Equal(t, "0", reply)
+
+		reply, err = e2eframework.SendGameServerUDP(gs, "PLAYER_CAPACITY 20")
+		if err != nil {
+			t.Fatalf("Could not message GameServer: %v", err)
+		}
+		assert.Equal(t, "ACK: PLAYER_CAPACITY 20\n", reply)
+
+		reply, err = e2eframework.SendGameServerUDP(gs, "PLAYER_CAPACITY")
+		assert.NoError(t, err)
+		assert.Equal(t, "20", reply)
```

Comment:
I think new line is needed here and in all other responses.
```suggestion
		assert.Equal(t, "20\n", reply)
```